multimodal/src/autogluon/multimodal/data/process_document.py

```diff
@@ -275,16 +279,56 @@ def process_one_sample(
             try:
```

Comment:
I think we can replace the fitz logics here totally with pdf2image as following
```
for per_col_name, per_col_image_features in document_features.items():
            try:
                # Process PDF documents.
                if feature_modalities[per_col_name] == DOCUMENT_PDF:
                    from pdf2image import convert_from_path

                    # Convert PDF to PIL images.
                    doc_images = convert_from_path(per_col_image_features[0])
                    if doc_images:
                        doc_image = doc_images[0].convert(image_mode)
                        words, normalized_word_boxes = self.get_ocr_features(per_col_image_features[0], doc_image)
                    else:
                        raise ValueError("Failed to convert PDF to images.")
                else:
                    # Process document image.
                    with PIL.Image.open(per_col_image_features[0]) as doc_image:
                        doc_image = doc_image.convert(image_mode)
                        words, normalized_word_boxes = self.get_ocr_features(per_col_image_features[0], doc_image)
            except ImportError as e:
                raise e
            except Exception as e:
                if self.missing_value_strategy.lower() == "zero":
                    logger.debug(f"Using a zero image due to '{e}'")
                    doc_image = PIL.Image.new(image_mode, (self.size, self.size), color=0)
                    doc_image = doc_image.convert(image_mode)
                    words = ""  # empty words
                    normalized_word_boxes = [self.pad_token_box]
                else:
                    raise e
 ```

Reply:
Thanks Zain for your contribution. We highly appreciate it. However due to license incompatibility, removing the usage of fitz will be more appropriate instead of creating direct/in-direct linkage to it

Reply:
Thank you for your contributions and comments; they are greatly appreciated. Due to licensing inconsistencies, I removed `PyMuPDF` and transitioned to `pdf2image` implementation. Detailed instructions on how to install Poppler on various platforms have been provided for your convenience. I hope this helps.